src/renderers/SymbolInstanceRenderer.ts

```diff
@@ -114,8 +114,8 @@ export default class SymbolInstanceRenderer extends SketchRenderer {
 
     const overridableLayers = extractOverrides(masterTree.layers);
 
-    const overrides = overridableLayers.reduce(function inject(
-      memo: FileFormat.OverrideValue[],
+    const overrides = await overridableLayers.reduce(async function inject(
```

Comment:
wouldn't
```ts
const overrides = (await Promise.all(overridableLayers.map((
  memo: Promise<FileFormat.OverrideValue | undefined>[],
  reference: Override
) => memo.concat(async () => {
  ...
})
, [])).filter(x => !!x)
```
be faster and possibly more understandable?

Reply:
Possibly. I haven't really checked what it does, since it was quite convoluted; I just applied mechanically promises where relevant. I'll look into it.

Reply:
I've tried giving a quick stab at this, but it's not as straightforward, because it has parts where it calls itself recursively with other `reduce`. 

Definitely it needs refactoring because it's horribly windy and convoluted, and probably could use being broken down in multiple more focused methods as well, but I'm not sure if this is the PR where it's best to tackle it.